pkg/sync/sync_context.go

```diff
@@ -818,11 +825,18 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 			}
 		}
 
+		shouldSkipDryRunOnMissingResource := func() bool {
+			// skip dry run on missing resource error for all application resources
+			if sc.skipDryRunOnMissingResource {
+				return true
+			}
+			return (task.targetObj != nil && resourceutil.HasAnnotationOption(task.targetObj, common.AnnotationSyncOptions, common.SyncOptionSkipDryRunOnMissingResource)) ||
+				sc.hasCRDOfGroupKind(task.group(), task.kind())
+		}
+
 		if err != nil {
 			switch {
-			case apierrors.IsNotFound(err) &&
-				((task.targetObj != nil && resourceutil.HasAnnotationOption(task.targetObj, common.AnnotationSyncOptions, common.SyncOptionSkipDryRunOnMissingResource)) ||
-					sc.hasCRDOfGroupKind(task.group(), task.kind())):
+			case apierrors.IsNotFound(err) && shouldSkipDryRunOnMissingResource():
 				// Special case for custom resources: if CRD is not yet known by the K8s API server,
 				// and the CRD is part of this sync or the resource is annotated with SkipDryRunOnMissingResource=true,
 				// then skip verification during `kubectl apply --dry-run` since we expect the CRD
```

Comment:
Does this comment now belong to the newly created `shouldSkipDryRunOnMissingResource` func above?

Reply:
Not really, it can stay inside case condition